inc/taxonomy-content-language.php

```diff
@@ -134,6 +134,56 @@ function wmf_add_default_content_language( int $post_ID ): void {
 	}
 }
 
+/**
+ * Filter posts by content language on post listings.
+ *
+ * This is used to ensure that posts having different current-language
+ * taxonomy are not displayed on the post listings.
+ *
+ * @param \WP_Query $query Query object to filter.
+ * @return void
+ */
+function wmf_filter_posts_by_content_language( WP_Query $query ) : void {
+
+	// Don't filter the admin, singular pages or non-main queries.
+	if ( is_admin() || ! $query->is_main_query() || is_singular() ) {
```

Comment:
Since the content language taxonomy is only registered for posts, you may be able to also return early if the query is for other post types.

Reply:
Excellent suggestions.